src/Symfony/Component/Cache/Adapter/TraceableAdapter.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+
+/**
+ * An adapter that collects data about all cache calls.
+ *
+ * @author Aaron Scherer <aequasi@gmail.com>
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class TraceableAdapter implements AdapterInterface
+{
+    private $pool;
+    private $calls = array();
+
+    public function __construct(AdapterInterface $pool)
+    {
+        $this->pool = $pool;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem($key)
+    {
+        $event = $this->start(__FUNCTION__, $key);
+        try {
+            $item = $this->pool->getItem($key);
+        } finally {
+            $event->end = microtime(true);
+        }
+        if ($item->isHit()) {
+            ++$event->hits;
+        } else {
+            ++$event->misses;
+        }
+        $event->result = $item->get();
+
+        return $item;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasItem($key)
+    {
+        $event = $this->start(__FUNCTION__, $key);
+        try {
+            $event->result = $this->pool->hasItem($key);
+        } finally {
+            $event->end = microtime(true);
+        }
+
+        if ($event->result) {
```

Comment:
thinking twice about this, I'd prefer removng this if/else now :) `$event->result` already allows to infer it, and it could be controversial to map true==hasItem to a hit. wdyt?